aat/src/aat/java/uk/gov/hmcts/ccd/definitionstore/tests/functional/elasticsearch/ElasticsearchImportDefinitionTest.java

```diff
@@ -0,0 +1,200 @@
+package uk.gov.hmcts.ccd.definitionstore.tests.functional.elasticsearch;
+
+import java.io.File;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.Matchers.hasKey;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import io.restassured.RestAssured;
+import io.restassured.builder.RequestSpecBuilder;
+import io.restassured.http.ContentType;
+import io.restassured.response.ValidatableResponse;
+import io.restassured.specification.RequestSpecification;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.ccd.definitionstore.tests.AATHelper;
+import uk.gov.hmcts.ccd.definitionstore.tests.BaseTest;
+
+class ElasticsearchImportDefinitionTest extends BaseTest {
+
+    private static final String DEFINITION_FILE = "src/resource/CCD_TEST_ES.xlsx";
+    private static final String DEFINITION_FILE_WITH_NEW_FIELD = "src/resource/CCD_TEST_ES_WithNewField.xlsx";
+
+    private static final String CASE_INDEX_NAME = "es_test_cases-000001";
+    private static final String CASE_INDEX_ALIAS = "es_test_cases";
+    private static final String LONG_FIELD_TYPE = "long";
+    private static final String TEXT_FIELD_TYPE = "text";
+    private static final String DATE_FIELD_TYPE = "date";
+    private static final String NUMBER_FIELD_TYPE = "double";
+    private static final String KEYWORD_FIELD_TYPE = "keyword";
+
+    protected ElasticsearchImportDefinitionTest(AATHelper aat) {
+        super(aat);
+    }
+
+    @BeforeAll
+    static void setUp() {
+        // stop execution of these tests if the env variable is not set
+        assumeTrue(null != System.getenv("ELASTICSEARCH_BASE_URI"), () -> "Ignoring Elasticsearch tests, variable ELASTICSEARCH_BASE_URI not set");
+    }
+
+    @AfterEach
+    void deleteIndex() {
+        // deletes the index
+        asElasticsearchApiUser()
+            .when()
+            .delete(CASE_INDEX_NAME)
+            .then()
+            .statusCode(200)
+            .body("acknowledged", equalTo(true));
```

Comment:
this if meant to cleanup so that next test starts from a clean state?
in that case, shouldn't we also delete the alias?

Reply:
can we make that method more generic, i.e. call it for example `tearDown` and have the following body:
```deleteIndex()
     deleteAlias()....
```

Reply:
done